jujugui/static/gui/src/app/user/user.js

```diff
@@ -30,6 +41,23 @@ const User = class User {
     this.sessionStorage = cfg.sessionStorage || sessionStorage;
     this.localStorage = cfg.localStorage || localStorage;
     this._external = cfg.externalAuth || null;
+    this.expiration = cfg.expiration || getExpirationDate();
+  }
+
+  get expiration() {
+    return new Date(this.sessionStorage.getItem('expiration'));
+  }
+
+  set expiration(val) {
+    this.sessionStorage.setItem('expiration', val);
+  }
+
+  _checkExpiration() {
```

Comment:
I feel this is badly named. The method doesn't just check the expiration, it also clears it?